From f91707dd22145481bff410da65f3f3ddfb027beb Mon Sep 17 00:00:00 2001
From: IAP Desktop Build <iap-desktop+build@google.com>
Date: Tue, 17 Aug 2021 16:39:02 +0200
Subject: [PATCH 5/5] Fix `throw` statement to avoid cascading exception

Throw proper exception when no handler has been registered
for a DCS sequence. This fixes an issue where throwing
an exception could cause another exception if the first
parameter has not been extracted yet.
---
 VtNetCore/XTermParser/XTermSequenceHandlers.cs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/VtNetCore/XTermParser/XTermSequenceHandlers.cs b/VtNetCore/XTermParser/XTermSequenceHandlers.cs
index 6a78250..9c406bd 100644
--- a/VtNetCore/XTermParser/XTermSequenceHandlers.cs
+++ b/VtNetCore/XTermParser/XTermSequenceHandlers.cs
@@ -1922,7 +1922,7 @@
                     .SingleOrDefault();
 
                 if (handler == null)
-                    throw new Exception("There are no sequence handlers configured for type DcsSequence with param0 = " + sequence.Parameters[0].ToString());
+                    throw new EscapeSequenceException("There are no sequence handlers configured for type DcsSequence " + sequence.ToString(), sequence);
 
                 handler.Handler(sequence, controller);
 
-- 
2.17.1.windows.2

